remote-api/src/main/java/org/alfresco/repo/web/scripts/rule/AbstractRuleWebScript.java

```diff
@@ -432,10 +442,40 @@ private Serializable convertValue(QName typeQName, Object propertyValue) throws
 
     protected void checkRule(Rule rule)
     {
+        List<Action> actions = getAllActions(rule);
+        blockPrivateActions(actions);
+        validateOutboundRuleActions(rule, actions);
+    }
+
+    private List<Action> getAllActions(Rule rule) {

```

Comment:
I really don't like having to use instanceof here, but I just don't have any other idea. I could try to add "getAllActions" to Action interface and implement it accordingly, but it seems to clash between CompositeAction and Action interfaces idea.